### PR TITLE
Setup User Signup Page

### LINKED ISSUE #18: [8] Users should be able to sign up
Create user signup page. This should include the /signup view and the users/signup template.

### PR DESCRIPTION
**Pull request by iTaylorCodes**
### What does this PR do?
- This Pull Request sets up the user signup feature
- This PR closes #18 
### Description of Task to be completed
- Setup /signup view. Setup error handling. Setup users/signup template. Add styles to signup page. Create tests for signup view and error handler.
### How should this be manually tested?
- Visit /signup, fill out signup form and confirm submission of new user into database.
### Video Example
https://user-images.githubusercontent.com/76837421/130182969-dcd60cf7-b0d6-48ae-9bf4-e1b33f0322e5.mov

### CODE REVIEW

**Comment by iTaylorCodes** [comment]
Checking the reason Codacy and Coveralls services are failing.

**Comment by iTaylorCodes** [comment]
Fixed some code patterns to increase Codacy grade. Seems coverage percent decrease is due to branching logic in error handling.

**Review by alexkayabula — CHANGES_REQUESTED** [review]
@iTaylorCodes write the respective tests to the code dropping test coverage.

**Comment by alexkayabula** [comment]
Hey Ian, I have realized that some of the lines dropping corresponding
coverage have no relevant tests.
It is recommended practise to have all code tested before we merge.Please
write the necessary tests.

Regards,

Alex

File: run.py [hunk 1]
```diff
@@ -14,4 +16,40 @@
...
+                first_name=form.first_name.data,
+                last_name=form.last_name.data,
+                image_url=form.image_url.data or User.image_url.default.arg
+            )
+            db.session.commit()
+
+            return redirect('/signup/favorites')
+        # If db errors, redirect back to try again
+        except exc.SQLAlchemyError as e:
+            print(f'ERROR AT LINE 51. ERROR TYPE: {type(e)}')
```
Comment (alexkayabula) [inline run.py]: @iTaylorCodes what does `ERROR AT LINE 51` mean ?

File: tests/test_user_views.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
+from unittest import TestCase
+from models import db, User, Favorites, Match
+from run import app
+
+app.config["WTF_CSRF_ENABLED"] = False
```
Comment (alexkayabula) [inline tests/test_user_views.py]: @iTaylorCodes move this to the config file.

File: models.py [hunk 3]
```diff
@@ -1,11 +1,11 @@
 from flask_bcrypt import Bcrypt
 from flask_sqlalchemy import SQLAlchemy
-from sqlalchemy.sql.elements import Null
```
Comment (alexkayabula) [inline models.py]: @iTaylorCodes how is removing this related to setting up sign up ?

**Review by alexkayabula — CHANGES_REQUESTED** [review]

File: tests/test_user_views.py (hunk 2 above)
Comment (iTaylorCodes) [reply tests/test_user_views.py]: Will do 👍

File: models.py (hunk 3 above)
Comment (iTaylorCodes) [reply models.py]: @alexkayabula Whoops, I removed it in an attempt to help Codacy coverage to pass

File: run.py (hunk 1 above)
Comment (iTaylorCodes) [reply run.py]: @alexkayabula Was printing error to server log, and giving it a location the error took place. I'll fix that 👍

**Comment by iTaylorCodes** [comment]
@alexkayabula I've updated the PR description with a video example.

**Review by alexkayabula — APPROVED** [review]
Great effort.

**Pull request merged**